### PR TITLE
drivers: spi: Fix DMA overflow in NXP MCUX driver

### PR DESCRIPTION
**Pull request by mmahadevan108**
The dummy buffer is not allocated to match the amount of data that is transferred. This results in DMA transfer overflowing to overwrite memory. This resulted in SPI test failures with DMA.

### CODE REVIEW

**Review by dleach02 — DISMISSED** [review]

File: drivers/spi/spi_mcux_flexcomm.c [hunk 1]
```diff
@@ -631,12 +631,19 @@ static int transceive_dma(const struct device *dev,
 
 		data->status_flags = 0;
 
+		if (data->ctx.rx_buf == NULL) {
+			dummy_rx_buffer = k_malloc(dma_len);
```
Comment (teburd) [inline drivers/spi/spi_mcux_flexcomm.c:635]: Shouldn't this not be needed if the DMA is setup to not increment the address of the receive when using the dummy word?

**Review by teburd — CHANGES_REQUESTED** [review]
This seems wrong, edma supports not-incrementing the address? This seems like maybe a misconfiguration when using the dummy rather than requiring a temporary buffer to receive into like this patch does

File: drivers/spi/spi_mcux_flexcomm.c (hunk 1 above)
Comment (mmahadevan108) [reply drivers/spi/spi_mcux_flexcomm.c]: Great point @teburd and thank you for your review comment.
On further investigation I did come across for a similar feature in this controller as well. I have updated the PR.

File: drivers/spi/spi_mcux_flexcomm.c [hunk 2]
```diff
@@ -519,6 +519,8 @@ static int spi_mcux_dma_rx_load(const struct device *dev, uint8_t *buf,
 	if (buf == NULL) {
 		/* if rx buff is null, then write data to dummy address. */
 		blk_cfg->dest_address = (uint32_t)&dummy_rx_buffer;
+		blk_cfg->source_addr_adj = DMA_ADDR_ADJ_NO_CHANGE;
```
Comment (teburd) [inline drivers/spi/spi_mcux_flexcomm.c:522]: This seems like the correct option regardless of whether the dummy or not is being used

**Review by teburd — CHANGES_REQUESTED** [review]

File: drivers/spi/spi_mcux_flexcomm.c (hunk 2 above)
Comment (mmahadevan108) [reply drivers/spi/spi_mcux_flexcomm.c]: I have removed this line as source increment is already 0 based on the transfer type (PERIPHERAL to MEMORY). I had to update the DMA driver as it was looking at `source and destination addr_adj` elements together.

**Review by teburd — APPROVED** [review]

File: drivers/spi/spi_mcux_flexcomm.c (hunk 2 above)
Comment (teburd) [reply drivers/spi/spi_mcux_flexcomm.c]: makes sense to me

**Review by dleach02 — APPROVED** [review]

**Pull request merged**